packages/core/src/sessionManagement.ts

```diff
@@ -89,25 +123,50 @@ function retrieveSession(sessionCookie: CookieCache): SessionState {
   return session
 }
 
-export function persistSession(session: SessionState, cookie: CookieCache) {
+export function persistSession(session: SessionState, cookie: CookieCache, withNewSessionStrategy = false) {
+  if (withNewSessionStrategy && !utils.isEmptyObject(session)) {
+    session.expire = String(Date.now() + SESSION_EXPIRATION_DELAY)
+  }
   const cookieString = utils
     .objectEntries(session)
     .map(([key, value]) => `${key}=${value}`)
     .join(SESSION_ENTRY_SEPARATOR)
-  cookie.set(cookieString, EXPIRATION_DELAY)
+  cookie.set(cookieString, SESSION_EXPIRATION_DELAY)
 }
 
 export function stopSessionManagement() {
-  registeredActivityListeners.forEach((e) => e())
-  registeredActivityListeners = []
+  registeredListeners.forEach((e) => e())
+  registeredIntervals.forEach((interval) => clearInterval(interval))
+  registeredListeners = []
+  registeredIntervals = []
 }
 
-let registeredActivityListeners: Array<() => void> = []
+let registeredListeners: Array<() => void> = []
+let registeredIntervals: number[] = []
 
 export function trackActivity(expandOrRenewSession: () => void) {
+  const doExpandOrRenewSession = monitor(expandOrRenewSession)
   const options = { capture: true, passive: true }
-  ;['click', 'touchstart', 'keydown', 'scroll'].forEach((event: string) => {
-    document.addEventListener(event, expandOrRenewSession, options)
-    registeredActivityListeners.push(() => document.removeEventListener(event, expandOrRenewSession, options))
+  ;[utils.DOM_EVENT.CLICK, utils.DOM_EVENT.TOUCH_START, utils.DOM_EVENT.KEY_DOWN, utils.DOM_EVENT.SCROLL].forEach(
+    (event: string) => {
+      document.addEventListener(event, doExpandOrRenewSession, options)
+      registeredListeners.push(() => document.removeEventListener(event, doExpandOrRenewSession, options))
+    }
+  )
+}
+
+function trackVisibility(expandSession: () => void, visibilityStateProvider: () => VisibilityState) {
+  const expandSessionWhenVisible = monitor(() => {
+    if (visibilityStateProvider() === 'visible') {
+      expandSession()
+    }
   })
+
+  const visibilityCheckInterval = window.setInterval(expandSessionWhenVisible, VISIBILITY_CHECK_DELAY)
+  document.addEventListener(utils.DOM_EVENT.VISIBILITY_CHANGE, expandSessionWhenVisible)
+
+  registeredIntervals.push(visibilityCheckInterval)
```

Comment:
I would rename `registeredListeners` to `stopCallbacks` or `stopSessionManagementCallbacks`, and use only one array:
```suggestion
  stopSessionManagementCallbacks.push(() => clearInterval(visibilityCheckInterval))
```